config/config.go

```diff
@@ -967,6 +972,15 @@ func (c *Config) LoadConfigData(data []byte) error {
 	return nil
 }
 
+func isASCII(s string) bool {
+	for i := 0; i < len(s); i++ {
+		if s[i] > unicode.MaxASCII {
```

Comment:
I think this will work. But, s[i] will give you a byte and unicode.MaxASCII is a rune. You might want to range over the string instead to view it rune by rune instead of byte by byte. As in: `for _, r := range s { if r > unicode.MaxASCII ...` But, this might still work anyway if it just needs to detect the first byte of the unicode being higher than MaxASCII. I'm not enough of an expert on Unicode to know for sure.

Reply:
when extended characters that are common to Nordic languages are used like `ø` or `Ø` it would be good to gracefully handle by automatically converting to the ["ASCII standard"](https://www.ascii-codes.com/cp865.html#standard_character_set) equivalent `o` or `O`, etc. 
In the event where a non linguistic character is used like ▒ or ╗ etc. then throw error

There are some interesting implementations of [RFC3490](https://datatracker.ietf.org/doc/html/rfc3490); see IDNA under [go-IDN](https://code.google.com/archive/p/go-idn/)

Reply:
@dhaifley good catch, but one of the clever hacks of UTF-8 is that in utf-8, ASCII characters are single-byte corresponding to the first 128 Unicode characters, so that's why it works here.

@JohannVonDiesel good point, I debated on whether to go down that path. I opted not to because I think systems should minimize the amount of "magic".  I'd rather tell the user that we don't accept non ASCII characters and have the client fail than have them wondering why the hostname in our system doesn't match the hostname on their system.